src/arrayf/arrayf.ts

```diff
@@ -32,23 +33,96 @@ export class arrayf {
      * Return value from array that generates the highest return value when
      * passed as a parameter to the score function.
      *
+     * Multiple score functions can be passed with later functions used only to
+     * break ties for values returned from the previous score function.
+     *
      * Example:
      * ```
      * const values = [{'a': 2, 'b': 3}, {'a': 1, 'b': 4}];
      * arrayf.max(values, (x) => x.a); // Returns {'a': 2, 'b': 3}
      * arrayf.max(values, (x) => x.b); // Returns {'a': 1, 'b': 4}
```

Comment:
Could you add an arrayf.spec file and add these as tests?

degu doesn't have tests for class that directly manipulate dom but for low level methods, I try to add a few tests.